### DISCUSSION TITLE
NUX Mighty Plug and audio device problems

### DISCUSSION THREAD

**Discussion by animaliko** [jamulussoftware/jamulus#2136] [Technical Support & Help]
Hello,
 
maybe you can help or have some experience, as I am trying to use the NUX MIghty Plug as audio for Jamulus on Windows10.
 
I see that the USB audio device for the MIghtly Plug (as I read in the NUX control panel),   is ASIO compatible (vesion 4.82.0) and has SR of 48k.
 
When starting Jamulus, I can choose to use:
- NUX Audio device
- ASIO4ALl v2. When using this one, which is the recommended from Jamulus, I get the error messaege that the audio device is incompatible since it does not support 48k. If I go to the settings of that audio device, no matter if I check the resampling option from 44.1 to 48k or not.  
 
So my questions are, do you know which audio device I should use? or how to solve the problem? 
Or do I need to setup something else in Windows like in the windows audio settings?

Not sure if having the audio device NUX audio works correctly, because even when having "Mute" chosen in the Personal Mix menu, I still hear my guitar when I play it, and it should not happen if I understand it correctly, as I should hear the sound comming from the server.
 
Thanks,
Oscar

**Comment by ghost** [comment]
@animaliko said: ... "no matter if I check the resampling option from 44.1 to 48k or not."

Resampling will introduce unacceptable latency. This may be the incompatibility also.
The USB audio device (physical hardware) Must be sampling at 48k.
Hopefully you will find a setting that works. (note that I don't use NUX)

**Reply by animaliko** [reply]
Thanks, still trying :)

**Comment by npostavs** [comment]
Do you have the right devices enabled in the ASIO4ALL window? See https://jamulus.io/wiki/Installation-for-Windows and https://jamulus.io/kb/2021/03/20/ASIO4ALL-Examples.html

You could also try KoordASIO instead of ASIO4ALL, it uses the newer WASAPI instead of ASIO4ALL's use of the older WDM so it could work better (or worse, but can't hurt to try anyway). https://github.com/koord-live/KoordASIO/releases

**Reply by animaliko** [reply]
Thanks, yes, I can only open the ASIO4ALL cinfg menu when I I do not have my NUX device connected to the USB port of my computer. When I do that and start Jamulus, I get a pop up window that no audio device was found and asks me to open the config menu, then I can choose other audio device, like speakers, ... I choose the NUX
As soon I want to start Jamulus with my NUX audio device connected in the USB port, I will get the error message that the ASIO4ALL  audio device does not support the sample rate of 48k, which is wrong, right? And goes back to the NUX audio device. Pretty weird.
I think the problem is that when I open the NUX control panel, I see that this device currently uses and supports 48k. But when I open the ASIO4ALL Config page, it tells me that the NUX  Stereo uses 2x44.1k, so something is wrong there. IS this because it is stereo? Check the image attached:
![control_setups](https://user-images.githubusercontent.com/95023331/143602791-72d0881e-b08a-47d4-a63e-85f8d4f97f00.jpg)

**Reply by npostavs** [reply]
Hmm, maybe the WDM driver for NUX only supports 44.1k? Also, the fact that it seems to have only output and no input in the ASIO4ALL window is not a promising sign.

**Reply by dcorson-ticino-com** [reply]
Forget ASIO4ALL.
Your device has a native ASIO driver, so use that.
If you hear yourself when muted that means that some kind of "direct audio" is enabled in your interface.  You need to turn that off, either a button on the interface or in the driver, I don't know.  
But the ASIO4ALL configuration will not help setting up the NUX.

**Reply by animaliko** [reply]
Thanks, yes, I am looking for the option of disabling the direct audio, but did not find it yet, so I am contacting the NUX support

**Comment by dcorson-ticino-com** [comment]
> Not sure if having the audio device NUX audio works correctly, because even when having "Mute" chosen in the Personal Mix menu, I still hear my guitar when I play it, and it should not happen if I understand it correctly, as I should hear the sound comming from the server.

Does the NUX have something like "direct audio" turned on.  Then you hear only what is coming from yourself and not that what is coming back from the server. 
I would try to get the NUX device to work as it should give you better latency than ASIO4ALL which I personally have never been able to get to work reliably.

**Reply by animaliko** [reply]
Thanks, yes, I am looking for the option of disabling the direct audio, but did not find it yet, so I am contacting the NUX support
